docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Extensions` are used by a particular group or several groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be installed and executed from `NEAR CLI`.
+- `Core NEAR CLI` extensions system only allows to introduce top-level commands through its design.
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
```

Comment:
hmm, what about `account -> view`? -- since the context is under `account` anyways, so no need to repeat it. Every other `view-*` would be a more specific form of this.

Reply:
> hmm, what about `account -> view`? -- since the context is under `account` anyways, so no need to repeat it. Every other `view-*` would be a more specific form of this.

I don't think we need to sacrifice readability for brevity in near-cli-rs since I believe that we should not expect people to type those commands anyway, and interactive mode proved to be working great.

> I'm not a fan of `view-account` here.

Can you share why? Is it ambiguous or misleading for you?

> Should we rename it to `view-account-info` or something simillar?

We certainly can. We currently name it `account-summary`, so it could also be `view-account-summary`

Reply:
I agree with @frol here. If this is in the interactive mode, I would be a fan of having more readable commands like `view-account-summary`.

Reply:
@frol because `view-account` is not descriptive enough. `view-account-summary` is ok.